remote-trait-object/src/port/client.rs

```diff
@@ -15,26 +15,113 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 use crate::packet::{Packet, PacketView, SlotId};
-use crossbeam::channel::{Receiver, Sender};
+use crate::queue::Queue;
+use crossbeam::channel::RecvTimeoutError::{Disconnected, Timeout};
+use crossbeam::channel::{bounded, Receiver, RecvError, Sender};
+use std::sync::Arc;
+use std::thread;
+use std::time;
 
+#[cfg(debug_assertions)]
+const TIMEOUT: std::time::Duration = std::time::Duration::from_millis(1_000_000);
+#[cfg(not(debug_assertions))]
+const TIMEOUT: std::time::Duration = std::time::Duration::from_millis(50);
+
+// FIXME: read from config
+const CALLSLOT_SIZE: u32 = 100;
+
+/// CallSlot represents an instance of call to the another module
+#[derive(Debug)]
+struct CallSlot {
+    id: SlotId,
+    response: Receiver<Packet>,
+}
+
+#[derive(Debug)]
 pub struct Client {
+    call_slots: Arc<Queue<CallSlot>>,
     ipc_send: Sender<Packet>,
-    ipc_recv: Receiver<Packet>,
+    receiver_thread: Option<thread::JoinHandle<()>>,
+    joined_event_receiver: Receiver<()>,
 }
 
 impl Client {
     pub fn new(ipc_send: Sender<Packet>, ipc_recv: Receiver<Packet>) -> Self {
+        let (joined_event_sender, joined_event_receiver) = bounded(1);
+        let callslot_size = SlotId::new(CALLSLOT_SIZE);
+        let call_slots = Arc::new(Queue::new(callslot_size.as_usize()));
+        let mut to_slot_receivers = Vec::with_capacity(callslot_size.as_usize());
+
+        for i in 0..callslot_size.as_raw() {
+            let (send_to_slot_recv, recv_for_slot) = bounded(1);
+            call_slots.push(CallSlot {
+                id: SlotId::new(i),
+                response: recv_for_slot,
+            });
+            to_slot_receivers.push(send_to_slot_recv);
+        }
+
         Client {
+            call_slots,
             ipc_send,
-            ipc_recv,
+            receiver_thread: Some(
+                thread::Builder::new()
+                    .spawn(move || {
+                        if let Err(RecvError) = receive_loop(ipc_recv, to_slot_receivers) {
+                            // Multiplexer is closed
+                        }
+                        joined_event_sender.send(()).unwrap();
+                    })
+                    .unwrap(),
+            ),
+            joined_event_receiver,
         }
     }
 
     pub fn call(&self, packet: PacketView) -> Packet {
-        // Please set call slot
-        let mut packet = packet.to_owned();
-        packet.set_slot(SlotId::new_request());
-        self.ipc_send.send(packet).unwrap();
-        self.ipc_recv.recv().unwrap()
+        let slot = self.call_slots.pop(Some(TIMEOUT)).expect("Module doesn't responde");
```

Comment:
`remote-trait-object` doesn't care about the concept `Module` anymore.
How about "Too many calls on port"?